tools/flang2/flang2exe/cgmain.cpp

```diff
@@ -14138,66 +14141,15 @@ reset_expr_id(void)
 static void
 store_return_value_for_entry(OPERAND *p, int i_name)
 {
-  TMPS *tmp = make_tmps();
-
-  if (p->ot_type != OT_VAR || !DT_ISCMPLX(DTYPEG(p->val.sptr))) {
-    print_token("\t");
-    print_tmp_name(tmp);
-    print_token(" = bitcast ");
-    write_type(make_generic_dummy_lltype());
-    print_token(" %");
-    print_token(get_entret_arg_name());
-    print_token(" to ");
-    write_type(make_ptr_lltype(p->ll_type));
-    print_nl();
-
-    print_token("\tstore ");
-    write_type(p->ll_type);
-    print_space(1);
-    write_operand(p, "", FLG_OMIT_OP_TYPE);
-    print_token(", ");
-
-    write_type(make_ptr_lltype(p->ll_type));
-    print_space(1);
-    print_tmp_name(tmp);
-    print_token(", align 4\n");
-  } else {
-    TMPS *loadtmp;
-    /*  %10 = bitcast i64* %__master_entry_rslt323 to <{float, float}>*  */
-    print_token("\t");
-    print_tmp_name(tmp);
-    print_token(" = bitcast ");
-    write_type(make_generic_dummy_lltype());
-    print_token(" %");
-    print_token(get_entret_arg_name());
-    print_token(" to ");
-    write_type(p->ll_type);
-    print_nl();
-
-    /*  %11 = load <{float, float}>, <{float, float}>* %cp1_300, align 4  */
-    loadtmp = make_tmps();
-    print_token("\t");
-    print_tmp_name(loadtmp);
-    print_token(" = load");
-    write_type(p->ll_type->sub_types[0]);
-    print_token(", ");
-    write_type(p->ll_type);
-    print_space(1);
-    write_operand(p, "", FLG_OMIT_OP_TYPE);
-    print_token(", align 4\n");
-
-    /*  store <{float, float}> %11, <{float, float}>* %10, align 4  */
-    print_token("\tstore ");
-    write_type(p->ll_type->sub_types[0]);
-    print_space(1);
-    print_tmp_name(loadtmp);
-    print_token(", ");
-
-    write_type(p->ll_type);
-    print_space(1);
-    print_tmp_name(tmp);
-    print_token(", align 4\n");
-  }
+  print_token("\tstore ");
+  write_type(p->ll_type);
+  print_space(1);
+  write_operand(p, "", FLG_OMIT_OP_TYPE);
```

Comment:
It seems that the new code no longer handles the case where `p->ot_type == OT_VAR`. Why?

Reply:
> It seems that the new code no longer handles the case where `p->ot_type == OT_VAR`. Why?

The case `p->ot_type == OT_VAR` does need to be handled, which was previously missed. I will update this PR to handle this case.

Reply:
Thank you. The fact that this case was missed, and yet the PR still passed CI jobs, indicates that we lack test coverage of this code. Would you be able to add a simple test case to exercise this path?

Reply:
> Thank you. The fact that this case was missed, and yet the PR still passed CI jobs, indicates that we lack test coverage of this code. Would you be able to add a simple test case to exercise this path?

I add a test case and try to cover these codes. But it fails because of some other reasons. I've filed an issue  #1336  and I don't think it is related with opaque pointers.

Reply:
OK thanks.